sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableInfo.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connector.catalog;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import com.google.common.collect.Maps;
+import org.apache.spark.sql.connector.expressions.Transform;
+import org.apache.spark.sql.types.StructType;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class TableInfo {
+
+  private final Column[] columns;
+  private final Map<String, String> properties;
+  private final Transform[] partitions;
+
+  /**
+   * Constructor for TableInfo used by the builder.
+   * @param builder Builder.
+   */
+  private TableInfo(Builder builder) {
+    this.columns = builder.columns;
+    this.properties = Collections.unmodifiableMap(builder.properties);
+    this.partitions = builder.partitions;
+  }
+
+  public Column[] columns() {
+    return columns;
+  }
+
+  public StructType schema() {
+    return CatalogV2Util.v2ColumnsToStructType(columns);
+  }
+
+  public Map<String, String> properties() {
+    return properties;
+  }
+
+  public Transform[] partitions() {
+    return partitions;
+  }
+
+  public static class Builder {
+    private Column[] columns;
+    private Map<String, String> properties;
+    private Transform[] partitions;
+
+    public Builder withColumns(Column[] columns) {
+      this.columns = columns;
+      return this;
+    }
+
+    public Builder withSchema(StructType schema) {
```

Comment:
no we don't need this. The builder is used by Spark and Spark should not use the deprecated StructType schema to create TableInfo.